diff --git a/vendor/amasty/module-extra-fee/Plugin/Klarna/ModuleKp/Model/Api/Request/BuilderPlugin.php b/vendor/amasty/module-extra-fee/Plugin/Klarna/ModuleKp/Model/Api/Request/BuilderPlugin.php
index ae33505..1c1e13b 100644
--- a/vendor/amasty/module-extra-fee/Plugin/Klarna/ModuleKp/Model/Api/Request/BuilderPlugin.php
+++ b/vendor/amasty/module-extra-fee/Plugin/Klarna/ModuleKp/Model/Api/Request/BuilderPlugin.php
@@ -45,4 +45,19 @@ class BuilderPlugin

         return [$data];
     }
+
+    /**
+     * Recollect totals for Klarna validation
+     *
+     * @param Builder $subject
+     * @param $amount
+     * @return int
+     */
+    public function beforeSetOrderAmount(Builder $subject, $amount)
+    {
+        $quote = $this->checkoutSession->getQuote();
+        $quote->collectTotals();
+
+        return (int) round($quote->getBaseGrandTotal()) * 100;
+    }
 }
