diff --git a/vendor/magento/module-google-tag-manager/Observer/SendCookieOnCartActionCompleteObserver.php b/vendor/magento/module-google-tag-manager/Observer/SendCookieOnCartActionCompleteObserver.php
index 390bf71..ae65833 100644
--- a/vendor/magento/module-google-tag-manager/Observer/SendCookieOnCartActionCompleteObserver.php
+++ b/vendor/magento/module-google-tag-manager/Observer/SendCookieOnCartActionCompleteObserver.php
@@ -13,6 +13,7 @@
 use Magento\Framework\Stdlib\Cookie\CookieMetadataFactory;
 use Magento\Framework\Stdlib\CookieManagerInterface;
 use Magento\GoogleTagManager\Model\Config\TagManagerConfig;
+use Magento\Framework\UrlInterface;
 
 /**
  * Observer for Cart Changes
@@ -51,6 +52,8 @@
      */
     protected $cookieMetadataFactory;
 
+    protected $_urlInterface;
+
     /**
      * @param \Magento\GoogleTagManager\Helper\Data $helper
      * @param Registry $registry
@@ -65,6 +68,7 @@
         CookieManagerInterface $cookieManager,
         Data $jsonHelper,
         CookieMetadataFactory $cookieMetadataFactory,
+        UrlInterface $urlInterface,
         RequestInterface $httpRequest
     ) {
         $this->helper = $helper;
@@ -72,6 +76,7 @@
         $this->cookieManager = $cookieManager;
         $this->jsonHelper = $jsonHelper;
         $this->cookieMetadataFactory = $cookieMetadataFactory;
+        $this->_urlInterface = $urlInterface;
         $this->request = $httpRequest;
     }
 
@@ -84,7 +89,7 @@
      */
     public function execute(Observer $observer)
     {
-        if (!$this->helper->isTagManagerAvailable()) {
+        if (!$this->helper->isTagManagerAvailable() || str_contains($this->_urlInterface->getCurrentUrl(), '/checkout/klarna/cookie/')) {
             return $this;
         }
         $productsToAdd = $this->registry->registry('GoogleTagManager_products_addtocart');
