diff --git a/vendor/magento/module-google-tag-manager/Observer/SetGoogleAnalyticsOnCheckoutObserver.php b/vendor/magento/module-google-tag-manager/Observer/SetGoogleAnalyticsOnCheckoutObserver.php
index 390bf71..ae65833 100644
--- a/vendor/magento/module-google-tag-manager/Observer/SetGoogleAnalyticsOnCheckoutObserver.php
+++ b/vendor/magento/module-google-tag-manager/Observer/SetGoogleAnalyticsOnCheckoutObserver.php
@@ -12,6 +12,7 @@
 use Magento\Framework\Serialize\SerializerInterface;
 use Magento\GoogleTagManager\Block\ListJson;
 use Magento\GoogleTagManager\Helper\Data;
+use Magento\Framework\UrlInterface;
 
 /**
  * Set Gtag on checkout observer
@@ -39,6 +40,8 @@
      */
     protected $scopeConfig;
 
+    protected $_urlInterface;
+
     /**
      * @param Data $helper
      * @param Session $checkoutSession
@@ -49,11 +52,13 @@
         Data $helper,
         Session $checkoutSession,
         SerializerInterface $jsonHelper,
+        UrlInterface $urlInterface,
         ScopeConfigInterface $scopeConfig
     ) {
         $this->helper = $helper;
         $this->checkoutSession = $checkoutSession;
         $this->jsonHelper = $jsonHelper;
+        $this->_urlInterface = $urlInterface;
         $this->scopeConfig = $scopeConfig;
     }
 
@@ -68,7 +73,7 @@
      */
     public function execute(Observer $observer)
     {
-        if (!$this->helper->isTagManagerAvailable()) {
+        if (!$this->helper->isTagManagerAvailable() || str_contains($this->_urlInterface->getCurrentUrl(), '/checkout/klarna/cookie/')) {
             return $this;
         }
         /** @var \Magento\Checkout\Controller\Onepage $controllerAction */
