diff --git a/vendor/scommerce/trackingbase/Model/GetProductPrice.php b/vendor/scommerce/trackingbase/Model/GetProductPrice.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/scommerce/trackingbase/Model/GetProductPrice.php
+++ b/vendor/scommerce/trackingbase/Model/GetProductPrice.php
@@ -133,7 +133,7 @@
                 }
             }
         }
-        return number_format($price, 2);
+        return number_format((float)$price, 2);
     }

     /**
