diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/AdsPixelSettings.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/AdsPixelSettings.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/AdsPixelSettings.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/AdsPixelSettings.php
@@ -151,7 +151,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -160,7 +160,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -170,7 +170,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -183,7 +183,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void{
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Content.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Content.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Content.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Content.php
@@ -205,7 +205,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -214,7 +214,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -224,7 +224,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -237,7 +237,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/CustomData.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/CustomData.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/CustomData.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/CustomData.php
@@ -358,7 +358,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -367,7 +367,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -377,7 +377,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -390,7 +390,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Event.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Event.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Event.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/Event.php
@@ -322,7 +322,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -331,7 +331,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -341,7 +341,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -354,7 +354,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventRequest.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventRequest.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventRequest.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventRequest.php
@@ -409,7 +409,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -418,7 +418,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -428,7 +428,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -441,7 +441,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventResponse.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventResponse.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventResponse.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/EventResponse.php
@@ -183,7 +183,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -192,7 +192,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -202,7 +202,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -215,7 +215,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

diff --git a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/UserData.php b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/UserData.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/UserData.php
+++ b/vendor/facebook/php-business-sdk/src/FacebookAds/Object/ServerSide/UserData.php
@@ -532,7 +532,7 @@
    * @param integer $offset Offset
    * @return boolean
    */
-  public function offsetExists($offset) {
+  public function offsetExists($offset): bool {
     return isset($this->container[$offset]);
   }

@@ -541,7 +541,7 @@
    * @param integer $offset Offset
    * @return mixed
    */
-  public function offsetGet($offset) {
+  public function offsetGet($offset): mixed {
     return isset($this->container[$offset]) ? $this->container[$offset] : null;
   }

@@ -551,7 +551,7 @@
    * @param mixed $value Value to be set
    * @return void
    */
-  public function offsetSet($offset, $value) {
+  public function offsetSet($offset, $value): void {
     if (is_null($offset)) {
       $this->container[] = $value;
     } else {
@@ -564,7 +564,7 @@
    * @param integer $offset Offset
    * @return void
    */
-  public function offsetUnset($offset) {
+  public function offsetUnset($offset): void {
     unset($this->container[$offset]);
   }

