diff --git a/vendor/facebook/facebook-for-magento2/Helper/MagentoDataHelper.php b/vendor/facebook/facebook-for-magento2/Helper/MagentoDataHelper.php
index 8204156867..9fdc0da1da 100644
--- a/vendor/facebook/facebook-for-magento2/Helper/MagentoDataHelper.php
+++ b/vendor/facebook/facebook-for-magento2/Helper/MagentoDataHelper.php
@@ -478,7 +478,7 @@
     }

     private function hashValue($string){
-        return hash('sha256', strtolower($string));
+        return hash('sha256', strtolower((string) $string));
     }

     // TODO Remaining user/custom data methods that can be obtained using Magento.
