diff --git a/vendor/auctane/api/Model/Action/Export.php b/vendor/auctane/api/Model/Action/Export.php
index 3cdba715f..f4c7b2e07 100644
--- a/vendor/auctane/api/Model/Action/Export.php
+++ b/vendor/auctane/api/Model/Action/Export.php
@@ -12,6 +12,7 @@ use Magento\Sales\Api\Data\OrderInterface;
 use Magento\Sales\Api\Data\OrderItemInterface;
 use Magento\Sales\Model\Order;
 use Magento\Sales\Model\ResourceModel\Order\Collection;
+use Amasty\Extrafee\Model\ResourceModel\ExtrafeeOrder\Collection as ExtraFee;
 
 
 /**
@@ -124,6 +125,9 @@ class Export
     /** @var WeightAdapter */
     private $weightAdapter;
 
+    /** @var ExtraFee */
+    private $extraFee;
+
     /**
      * Export class constructor
      *
@@ -134,6 +138,7 @@ class Export
      * @param \Auctane\Api\Helper\Data $dataHelper helper object
      * @param \Magento\GiftMessage\Helper\Message $giftMessage The gift message.
      * @param WeightAdapter $weightAdapter
+     * @param ExtraFee $extraFee
      */
     public function __construct(
         \Magento\Sales\Model\ResourceModel\Order\CollectionFactory $order,
@@ -142,7 +147,8 @@ class Export
         \Magento\Eav\Model\Config $eavConfig,
         \Auctane\Api\Helper\Data $dataHelper,
         \Magento\GiftMessage\Helper\Message $giftMessage,
-        WeightAdapter $weightAdapter
+        WeightAdapter $weightAdapter,
+        ExtraFee $extraFee
     )
     {
         $this->orderCollectionFactory = $order;
@@ -181,6 +187,7 @@ class Export
         $this->_typeBundle = Type::TYPE_BUNDLE;
 
         $this->weightAdapter = $weightAdapter;
+        $this->_extraFee = $extraFee;
     }
 
     /**
@@ -284,6 +291,15 @@ class Export
         $this->addXmlElement("InternalNotes", "<![CDATA[{$order->getCustomerNote()}]]>");
         $this->addXmlElement("StoreCode", $order->getStore()->getCode());
 
+        try {
+            $feeAmount = $this->_extraFee->addFilterByOrderId($order->getEntityId())->fetchItem();
+            $this->addXmlElement("CustomField1", $feeAmount->getTotalAmount() ?? '0.00');
+        } catch (\Magento\Framework\Exception\NoSuchEntityException $e) {
+            $this->addXmlElement("CustomField1", '0.00');
+        } catch (\Error $e) {
+            $this->addXmlElement("CustomField1", '0.00');
+        }
+
         $this->_getGiftMessageInfo($order);
 
         $this->_xmlData .= "\t<Customer>\n";
